src/hotspot/share/services/mallocHeader.hpp

```diff
@@ -129,10 +129,14 @@ class MallocHeader {
   // an option pointer to the corruption in p_corruption, and return false.
   // Return true if block is fine.
   inline bool check_block_integrity(char* msg, size_t msglen, address* p_corruption) const;
+  // Check correct alignment and placement of pointer, fill in short descriptive text and return false
+  // if this is not the case.
+  // Returns true if the memblock looks OK.
+  inline static bool check_pointer_integrity(uintptr_t header, char* msg, size_t msglen, address* p_corruption);
```

Comment:
This now cuts out functionality from MallocTracker::print_pointer_information(), which had been using check_block_integrity to make sure a pointer is safe to print. Now it resolves the header right away without checking the pointer.